### PR TITLE
Respect `ignore_tables` in Postgres structure dump

### PR DESCRIPTION
**Pull request by rustygeldmacher**
When using `sql` as the schema format, or even just doing `rake
db:structure:dump`, it would be good to respect the list of ignored
tables that has been configured.

### Summary

Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

If you are updating any of the CHANGELOG files or are asked to update the
CHANGELOG files by reviewers, please add the CHANGELOG entry at the top of the file.

Finally, if your pull request affects documentation or any non-code
changes, guidelines for those changes are [available
here](http://guides.rubyonrails.org/contributing_to_ruby_on_rails.html#contributing-to-the-rails-documentation)

Thanks for contributing to Rails!

### CODE REVIEW

File: activerecord/test/cases/tasks/postgresql_rake_test.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-require "cases/helper"
+any "cases/helper"
```
Comment (eileencodes) [inline activerecord/test/cases/tasks/postgresql_rake_test.rb]: Why did this change?

File: activerecord/lib/active_record/tasks/postgresql_database_tasks.rb [hunk 2]
```diff
@@ -66,6 +66,12 @@ def structure_dump(filename, extra_flags)
             "--schema=#{part.strip}"
           end
         end
+
+        ignore_tables = ActiveRecord::SchemaDumper.ignore_tables
+        if ignore_tables.any?
+          args += ignore_tables.flat_map { |table| ['-T', table] }
```
Comment (eileencodes) [inline activerecord/lib/active_record/tasks/postgresql_database_tasks.rb]: Can you change this to use double quotes?

**Review by eileencodes — CHANGES_REQUESTED** [review]
Added a question and a change requested by the linter.

File: activerecord/test/cases/tasks/postgresql_rake_test.rb (hunk 1 above)
Comment (rustygeldmacher) [inline activerecord/test/cases/tasks/postgresql_rake_test.rb]: Whoops, typo. Thanks for catching.

File: activerecord/lib/active_record/tasks/postgresql_database_tasks.rb (hunk 2 above)
Comment (rustygeldmacher) [inline activerecord/lib/active_record/tasks/postgresql_database_tasks.rb]: Will do!

**Comment by rustygeldmacher** [comment]
@eileencodes  -- Made those changes, thanks!

**Comment by maclover7** [comment]
@rustygeldmacher Are any of the other database adapters affected by this? If so, would you be willing to make another PR to fix them as well?

**Comment by yawboakye** [comment]
@rustygeldmacher Good stuff! @maclover7 there should be similar updates to the MySQL adapter. I'm not sure if it should be replicated in the SQLite3 adapter too. Here's the documentation for how to ignore tables in MySQL https://dev.mysql.com/doc/refman/5.7/en/mysqldump.html#option_mysqldump_ignore-table

**Comment by yawboakye** [comment]
@rustygeldmacher Your changes have been included in https://github.com/rails/rails/pull/29077

**Comment by guilleiguaran** [comment]
Closing this in favor of #29077.

@rustygeldmacher don't worry your changes will be included and you'll receive credits in the CHANGELOG and [Rails contributors app](http://contributors.rubyonrails.org/).

**Comment by rustygeldmacher** [comment]
Thanks @guilleiguaran -- looking forward to it being released!

**Pull request closed without merging**